usage/camp/src/test/java/org/apache/brooklyn/camp/brooklyn/catalog/CatalogYamlAppTest.java

```diff
@@ -24,50 +24,32 @@
 
 
 public class CatalogYamlAppTest extends AbstractYamlTest {
-    
-    /**
-     * "Contrived" example was encountered by a customer in a real use-case!
-     * I couldn't yet simplify it further while still reproducing the failure.
-     * Throws StackOverlfowError, without giving a nice error message about 
-     * "BasicEntity" cyclic reference.
-     * 
-     * The circular reference comes from the member spec referencing 
-     * "org.apache.brooklyn.entity.stock.BasicEntity", but that has been defined in the
-     * catalog as this new blueprint (which overrides the previous value of it
-     * being a reference to the Java class).
-     * 
-     * We need to use an id that matches something else already on the classpath.
-     * Otherwise we'd get an error telling us "could not resolve item ..." when
-     * attempting to add the initial catalog item.
-     */
-    @Test(groups="WIP") // TODO Fix this!
-    public void testAddCatalogItemWithCircularReference() throws Exception {
-        // Add a catalog item with a circular reference to its own id.
+
+    @Test
+    public void testAddCatalogItemWithMemberSpecCircularReference() throws Exception {
```

Comment:
Don't think so, the problem is in crossing the `memberSpec` boundary, so the changes are just to make the test minimal - being a child is not a factor which triggers the bug.
Don't mind it though, so brought back the original test + note.